project/SparkBuild.scala

```diff
@@ -560,6 +570,14 @@ object Unidoc {
   )
 }
 
+object Java8TestSettings {
+  import BuildCommons._
+
+  lazy val settings = Seq(
+    compilerJVMVersion := "1.8"
```

Comment:
Subtlety: this setting will break the `java8` tests when running on Scala 2.10.x, which does not support the emission of Java 8 bytecode. For this reason, we need separate settings for configuring the scalc and javac target JVM versions and need to only set scalac to 1.8 when running Scala 2.11.4 or higher.